tensorboard/util/test_case.py

```diff
@@ -0,0 +1,113 @@
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""TensorBoard test case helper module.
+
+This module provides a TensorBoard base test class with some of the
+niceties of tf.test.TestCase, while only requiring standard unittest
+be installed.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import atexit
+import os
+import shutil
+import tempfile
+import unittest
+
+from tensorboard.util import tb_logging
+
+
+logger = tb_logging.get_logger()
+
+_temp_dir = None
+
+
+def get_temp_dir():
+  """Return a temporary directory for tests to use."""
+  global _temp_dir
+  if not _temp_dir:
+    if os.environ.get('TEST_TMPDIR'):
+      temp_dir = tempfile.mkdtemp(prefix=os.environ['TEST_TMPDIR'])
+    else:
+      temp_dir = tempfile.mkdtemp()
+
+    def delete_temp_dir(dirname=temp_dir):
+      try:
+        shutil.rmtree(dirname)
+      except OSError as e:
+        logger.error('Error removing %s: %s', dirname, e)
+
+    atexit.register(delete_temp_dir)
+    _temp_dir = temp_dir
+
+  return _temp_dir
+
+
+class TestCase(unittest.TestCase):
+  """TensorBoard base test class.
+  This class can lazily create a temporary directory for tests to use.
+  """
+
+  def __init__(self, method='runTest'):
+    super(TestCase, self).__init__(method)
+    self._method = method
+    self._tempdir = None
+
+  def setUp(self):
+    super(TestCase, self).setUp()
+
+  def tearDown(self):
+    super(TestCase, self).tearDown()
+
+  def assertItemsEqual(self, actual, expected, msg=None):
```

Comment:
If you want to include this method, please prefer to delegate to
`six.assertCountEqual` in the implementation rather than checking
`hasattr`.

Reply:
I think I had some issue with delegating to six, but let me double check. That was the  original implementation.

Reply:
Actually, I think calling `six.assertCountEqual` caused a cycle.

Reply:
You’ll need to pass up the `super` object:

```py
    return six.assertCountEqual(super(TestCase, self), actual, expected, msg)
```

(Tested on all the `support_notf` tests on Python 2.7.)